scalalib/src/mill/scalalib/CoursierModule.scala

```diff
@@ -137,6 +138,15 @@ object CoursierModule {
         deps: IterableOnce[T],
         sources: Boolean = false
     ): Agg[PathRef] = {
+      // TODO: `getOrThrow` swallows error messages when Failure is returned.
+      // so I introduced resolveDeps0 where it makes sense to propagate the error message
+      resolveDeps0(deps, sources).getOrThrow
+    }
+
+    def resolveDeps0[T: CoursierModule.Resolvable](
+        deps: IterableOnce[T],
+        sources: Boolean = false
+    ): Result[Agg[PathRef]] = {
```

Comment:
This was fixed by adding special handling for `Result.Exception(Result.Failure())` when we show the error messages here: https://github.com/com-lihaoyi/mill/pull/3391
If you rebase on top of the current `main` branch you should not need to add `resolveDeps0`

Reply:
oh cool!